lib/Dialect/TritonGPU/Transforms/LoopDCE.cpp

```diff
@@ -0,0 +1,83 @@
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Pass/Pass.h"
+#include "triton/Dialect/TritonGPU/Transforms/Utility.h"
+
+using namespace mlir;
+using namespace triton;
+
+//===----------------------------------------------------------------------===//
+// recursiveLoopDCE
+//===----------------------------------------------------------------------===//
+
+// Recursively remove dead operations in the loop body. These are subgraphs of
+// memory-effect-free ops in the loop whose results are transitively only used
+// by themselves.
+static void recursiveLoopDCE(scf::ForOp loop) {
```

Comment:
Is it any different than this pattern:
https://github.com/triton-lang/triton/blob/main/lib/Dialect/TritonGPU/Transforms/Utility.cpp#L1089

Reply:
Ah, I didn't know this existed. I think the pattern already has propagation through `scf.if` that I was planning to implement as well! I thin I can delete this pass